addons/main/functions/fnc_jammerPlayerLocal.sqf

```diff
@@ -77,28 +77,65 @@ private _distRad = -1;
+		private _rxInterference = 1;
+		private _txInterference = 1;
+
+		// for now staying with linear degradation of signal. Might make it a tad better for players than the sudden commms -> no comms exponential could induce
+		private _rxInterference = _jamStrength - (_distPercent * _jamStrength) + 1; 	// recieving interference. above 1 to have any effect.
+		private _txInterference = 1 / _rxInterference; 									// transmitting interference, below 1 to have any effect.
+
+		// Set the TF receiving and sending distance multipliers
+		player setVariable ["tf_receivingDistanceMultiplicator", _rxInterference];
+		player setVariable ["tf_sendingDistanceMultiplicator", _txInterference];
+
+		//Debugging
+		// if (false) then {	
+		// 	// systemChat format ["Distance: %1, Percent: %2", _distJammer,  100 * _distPercent];
+		// 	systemChat format ["tfar_rx: %1, tfar_tx: %2", _rxInterference, _txInterference];
+		// 	systemChat format ["Closest Jammer netID: %1, radius: %2, enabled: %3", netId (_nearestJammer select 0), _nearestJammer select 1, _nearestJammer select 3];
+		// };
+	};
 };
 
-// we now got distance, and nearest jammer, time to calculate jamming
-private _distPercent = _distJammer / _distRad;
-private _jamStrength = _nearestJammer select 2;
-private _rxInterference = 1;
-private _txInterference = 1;
-
-// for now staying with linear degradation of signal. Might make it a tad better for players than the sudden commms -> no comms exponential could induce
-private _rxInterference = _jamStrength - (_distPercent * _jamStrength) + 1; 	// recieving interference. above 1 to have any effect.
-private _txInterference = 1 / _rxInterference; 									// transmitting interference, below 1 to have any effect.
-
-// Set the TF receiving and sending distance multipliers
-player setVariable ["tf_receivingDistanceMultiplicator", _rxInterference];
-player setVariable ["tf_sendingDistanceMultiplicator", _txInterference];
-
-//Debugging
-// if (false) then {	
-// 	// systemChat format ["Distance: %1, Percent: %2", _distJammer,  100 * _distPercent];
-// 	systemChat format ["tfar_rx: %1, tfar_tx: %2", _rxInterference, _txInterference];
-// 	systemChat format ["Closest Jammer netID: %1, radius: %2, enabled: %3", netId (_nearestJammer select 0), _nearestJammer select 1, _nearestJammer select 3];
-// };
\ No newline at end of file
+// handle drone jammers
+private _PP_film = GVAR(FilmGrain_jamEffect);
+_PP_film ppEffectEnable false; // restore players view back to normal if no other logic decides otherwise
+private _drone = getConnectedUAV player;
+if (!isNull _drone) then {
+	// sort drone jammers by distance to drone not distance to player (which only makes sense for VoiceCommJammers)
+	private _sortingCode = { _input0 distance (_x select 0) };	// sort by distance between drone and jammer
+	private _filterCode = { (_x select 3)  && ("DroneJammer" in (_x select 4)) };	// keep jammers that are enabled and have the "DroneJammer" capability
```

Comment:
When I began working on crowsEW I wanted to keep it consistent with how you did it.
Then when I learned that `#` has a higher operator preference than `select` I started using it more because you safe on a lot of `()` which you don't need anymore.

Reply:
Fair! Yeah, I believe I wrote a fair bit of this code before I learned of ``#`` so my use is a bit of a mix. I don't have a great code-style standard applied to this mod as I learned stuff as I developed it, and didn't refactor old stuff specifically 😅

Reply:
> You could also consider doing lazy eval for the if-statement, to avoid an array `in` command if the jammer is not enabled anyway.
> 
> ```sqf
> private _filterCode = { (_x select 3)  && { ("DroneJammer" in (_x select 4)) } };
> ```

Done in 05e235a4b262bd49f46bb463dd854406dc089bfa